header-chain/src/mmr_guest.rs

```diff
@@ -47,6 +47,10 @@ impl MMRGuest {
         }
         println!("GUEST: calculated subroot: {:?}", current_hash);
         println!("GUEST: subroots: {:?}", self.subroots);
-        self.subroots[mmr_proof.subroot_idx] == current_hash
+        if mmr_proof.subroot_idx < self.subroots.len() {
+            self.subroots[mmr_proof.subroot_idx] == current_hash
+        } else {
+            false
+        }
```

Comment:
It's probably cleaner to use [`get`](https://doc.rust-lang.org/std/primitive.slice.html#method.get) instead.

Reply:
Also, is it clear that this case can actually arise?